server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java

```diff
@@ -162,18 +178,37 @@ public List<JdbcGeneralResult> doInStatement(Statement statement) throws SQLExce
             return this.sqls.stream().map(sqlTuple -> {
                 JdbcGeneralResult result = JdbcGeneralResult.canceledResult(sqlTuple);
                 result.setConnectionReset(true);
+                if (context != null) {
+                    context.addResult(result);
+                }
+                listeners.forEach(
```

Comment:
we may use try-catch block to wrap this~

Reply:
done